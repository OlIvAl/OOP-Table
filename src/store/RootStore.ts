import { FinanceActionsStore } from './Actions/FinanceActionsStore';
import { IFinanceActionModel } from './Actions/interfaces';
import { IStoreWithCurrentModel } from './Abstract/interfaces';

export class RootStore {
  financeActionsStore: IStoreWithCurrentModel<IFinanceActionModel>;

  constructor() {
    this.financeActionsStore = new FinanceActionsStore(this);
  }
}