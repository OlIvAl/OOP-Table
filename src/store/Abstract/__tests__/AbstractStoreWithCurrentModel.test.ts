import { IStoreWithCurrentModel } from '../interfaces';
import { ITestModel } from '../Test/interfaces';
import TestStoreWithCurrentModel from '../Test/TestStoreWithCurrentModel';
import TestModel from '../Test/TestModel';
import { RootStore } from '../../RootStore';

const rootStore = new RootStore();
let store: IStoreWithCurrentModel<ITestModel>;
let models: ITestModel[];

beforeEach(() => {
  store = new TestStoreWithCurrentModel(rootStore);
  models = [new TestModel(store), new TestModel(store), new TestModel(store)];
  store.models = models;
});

describe('AbstractStore', () => {
  describe('constructor', () => {
    it('init valid store', () => {
      expect(store).toMatchSnapshot();
    });
    it('currentModel is null after init', () => {
      expect(store.currentModel).toBeNull();
    });
  });
  describe('setCurrentModel', () => {
    it('set existing model', () => {
      store.setCurrentModel(models[1]);

      expect(store.currentModel).toEqual(models[1]);
    });
    it('throw error when try to set not existing model', () => {
      expect(() => store.setCurrentModel(new TestModel(store))).toThrowError();
    });
  });
  describe('resetStore', () => {
    beforeEach(() => {
      store.currentModel = models[1];
    });

    it('clean models', () => {
      expect(store.models.length).toEqual(3);
      store.resetStore();
      expect(store.models.length).toEqual(0);
    });
    it('set currentModel in null', () => {
      expect(store.currentModel).toEqual(models[1]);
      store.resetStore();
      expect(store.currentModel).toEqual(null);
    });
  });
  describe('resetCurrentModel', () => {
    beforeEach(() => {
      store.currentModel = models[1];
    });

    it('not clean models', () => {
      expect(store.models.length).toEqual(3);
      store.resetCurrentModel();
      expect(store.models.length).toEqual(3);
    });
    it('set currentModel in null', () => {
      expect(store.currentModel).toEqual(models[1]);
      store.resetCurrentModel();
      expect(store.currentModel).toEqual(null);
    });
  });
});
