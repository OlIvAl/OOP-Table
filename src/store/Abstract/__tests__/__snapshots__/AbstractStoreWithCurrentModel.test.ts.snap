// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractStore constructor init valid store 1`] = `
TestStoreWithCurrentModel {
  "currentModel": null,
  "models": Array [
    TestModel {
      "store": [Circular],
    },
    TestModel {
      "store": [Circular],
    },
    TestModel {
      "store": [Circular],
    },
  ],
  "rootStore": RootStore {
    "financeActionsStore": FinanceActionsStore {
      "rootStore": [Circular],
    },
  },
}
`;
